lib/using_restricted_array.rb

```diff
@@ -6,56 +6,114 @@
 
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) because I'm performing a linear search until the loop reaches nil in the array.
+# Space complexity: O(1), because I'm only looking for 1 value, it doesn't matter what size the array is.
 def length(array)
-  raise NotImplementedError
+  i = 0
+  until array[i] == nil
+      i += 1
+  end
+  return i
 end
 
+
 # Prints each integer values in the array
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(1) because retrieving an element is constant, regardless of the size.
+# Space complexity: O(1) because size of the array doesn't matter? Still retrieving one element at a time.
 def print_array(array)
```

Comment:
👍 , however the time complexity is O(n) because you print out `n` elements and it increases as the size of the array increases proportionately.